bftengine/include/bftengine/ReplicaConfigSingleton.hpp

```diff
@@ -0,0 +1,51 @@
+//Concord
+//
+//Copyright (c) 2019 VMware, Inc. All Rights Reserved.
+//
+//This product is licensed to you under the Apache 2.0 license (the "License").  You may not use this product except in compliance with the Apache 2.0 License. 
+//
+//This product may include a number of subcomponents with separate copyright notices and license terms. Your use of these subcomponents is subject to the terms and conditions of the subcomponent's license, as noted in the LICENSE file.
+
+#pragma once
+
+#include <stdint.h>
+#include <set>
+#include <string>
+
+#include "ReplicaConfig.hpp"
+
+class IThresholdSigner;
+class IThresholdVerifier;
+
+namespace bftEngine {
+
+class ReplicaConfigSingleton {
+    public:
+	static ReplicaConfigSingleton& GetInstance(const ReplicaConfig* config=nullptr);
+
+	uint16_t GetFVal();
```

Comment:
All these getters should be const methods. Additionally, the methods that return pointers should return const* to prevent global mutation.